indexing-service/src/main/java/org/apache/druid/indexing/common/task/CompactionTask.java

```diff
@@ -452,6 +454,45 @@ public boolean isPerfectRollup()
     return tuningConfig != null && tuningConfig.isForceGuaranteedRollup();
   }
 
+  /**
+   * In MSQ, MVDs need conversion to array for rollup, and they aren't supported as partition columns. This check
+   * therefore is particularly for MSQCompactionRunner; it returns true when dimension types aren't known
+   * from CompactionTask spec, or if either rollup or partition dimensions contain any string-type column.
```

Comment:
```suggestion
   * Checks if multi-valued string dimensions need to be analyzed by downloading the segments.
   * This method returns true only for MSQ engine when either of the following holds true:
   * <ul>
   * <li> Range partitioning is done on a string dimension or an unknown dimension
   * (since MSQ does not support partitioning on a multi-valued string dimension) </li>
   * <li> Rollup is done on a string dimension or an unknown dimension
   * (since MSQ requires multi-valued string dimensions to be converted to arrays for rollup) </li>
   * </ul>
   * @return false for native engine, true for MSQ engine only when partitioning or rollup is done on a string or unknown dimension.
```